contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -1,91 +1,71 @@
 #include "contrib/hyperscan/matching/input_matchers/source/matcher.h"
 
-#include <numeric>
-
 namespace Envoy {
 namespace Extensions {
 namespace Matching {
 namespace InputMatchers {
 namespace Hyperscan {
 
-ScratchThreadLocal::ScratchThreadLocal(const hs_database_t* database) {
+ScratchThreadLocal::ScratchThreadLocal(const hs_database_t* database,
+                                       const hs_database_t* som_database) {
   hs_error_t err = hs_alloc_scratch(database, &scratch_);
   if (err != HS_SUCCESS) {
     throw EnvoyException(fmt::format("unable to allocate scratch space, error code {}.", err));
   }
+  if (som_database) {
+    err = hs_alloc_scratch(som_database, &scratch_);
+    if (err != HS_SUCCESS) {
+      throw EnvoyException(
```

Comment:
unreachable?

use IS_ENVOY_BUG?

Reply:
No, it is not unreachable. A malformed regex pattern will cause the exception.

Reply:
Oh, I see the incorrect line. Yes, it is unreachable as expected, and only will happen on lacking memories.

BTW, should we use `IS_ENVOY_BUG` so frequently? The exception will be thrown in config time but not runtime, and lacking memories is not a Envoy bug I think.

Reply:
Yeah I think what I really want to express here is that the line of code can't be hit.

There's certainly a concern with lines of code that can't be tested, and whether anyone would catch this assert successfully and what they would do about that if there was no memory left.

We used to have a macro to indicate specifically that a line of code can't be hit, and previously the toolchain running the coverage analysis would not count those lines, but that stopped working at some point when the toolchain evolved.

WDYT of just using RELEASE_ASSERT here rather than throwing an exception?

Reply:
Then I think it will be fine to use `IS_ENVOY_BUG`, I will change all these unreachable exceptions to `IS_ENVOY_BUG`.

BTW, this is a contrib extension, and the CI would not cover it.